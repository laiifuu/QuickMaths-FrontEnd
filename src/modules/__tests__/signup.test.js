import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import store from '../../redux/configureStore';
import Signup from '../user-sessions/signup';

describe('Signup', () => {
  test('Signup page renders as expected', async () => {
    const snapHomepage = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
        ,
      </Provider>,
    )
      .toJSON();
    expect(snapHomepage).toMatchSnapshot();
  });
});
