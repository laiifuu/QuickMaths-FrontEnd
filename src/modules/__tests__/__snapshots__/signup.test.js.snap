// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Signup page renders as expected 1`] = `
Array [
  <section
    className="user-page flex"
  >
    <div>
      <h1>
        Get Started
      </h1>
    </div>
    <form
      action=""
      className="user-form flex"
    >
      <input
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="Username"
        type="input"
      />
      <div
        className="backend-error"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <p />
      </div>
      <div
        className="error"
        style={
          Object {
            "display": "none",
          }
        }
      >
        <p />
      </div>
      <button
        className="session-btn"
        name="signup"
        onClick={[Function]}
        type="button"
      >
        Sign Up
      </button>
      <a
        href="/user/login"
        onClick={[Function]}
      >
        <p
          className="session-redirect"
        >
          <em>
            Already a member? Log in...
          </em>
        </p>
      </a>
    </form>
  </section>,
  ",",
]
`;
